lib/multiaddr.js

```diff
@@ -17,7 +17,10 @@ export function multiaddrToHttpUrl (addr) {
         new Error(`Cannot parse "${addr}": unsupported http path`, { cause: err }),
         { code: 'INVALID_HTTP_PATH' }
       )
-    }
+    } // Handle DNS-based direct http/https format
+  } else if (hostType.startsWith('dns') && (multiAddrParts[0] === 'http' || multiAddrParts[0] === 'https')) {
```

Comment:
Can we make this code generic to support other host types than `dns`? I remember that many SPs use IPv4 address instead. Take a look at the function `getUriHost` to see all the different host types to support.

An example test case to add:

```js
  ['/ip4/127.0.0.1/http', 'http://127.0.0.1'],
```

Suggested fix:

```suggestion
    } 
  // Handle HTTP/HTTPs addresses using the default port
  } else if (multiAddrParts[0] === 'http' || multiAddrParts[0] === 'https') {
```

Reply:
Addressed in b71f5fc